project-schema.json: add ConservationMeasure and conservationMeasure.csv

odscjen opened the pull request:
**Related issues**

closes #401 

**Description**

<!-- If the changes in the PR are not sufficiently explained by the related issues and commit messages, add a description here. -->

**Merge checklist**

<!-- Complete the checklist before requesting a review. -->

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [ ] `docs/examples/example.json`
  - [ ] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [x] Add an entry to `docs/reference/codelists.md`

odscjen committed:
project-schema.json: add ConservationMeasure and conservationMeasure.csv

odscjen committed:
update changelog.md

duncandewhurst commented on schema/project-level/project-schema.json:
```diff
@@ -352,6 +352,17 @@
+            "$ref": "#/definitions/ConservationMeasure"
+          },
+          "uniqueItems": true,
+          "minItems": 1,
+          "wholeListMerge": true
```
I think we'll need to remove this, depending on the resolution of https://github.com/open-contracting/infrastructure/pull/434#discussion_r1401173099

```suggestion
          "minItems": 1
```

duncandewhurst commented on docs/reference/codelists.md:
```diff
@@ -83,6 +83,13 @@ Projects with a `status` of 'completed' may be displayed in a list of archived p
 
 ## Open codelists
 
+### conservationMeasure
```
In keeping with the other headings.

```suggestion
### ConservationMeasure
```

duncandewhurst commented on schema/project-level/project-schema.json:
```diff
@@ -2370,6 +2381,27 @@
+        "description": {
+          "title": "Description",
+          "description": "A short description of the conservation measure.",
+          "type": "string"
```
```suggestion
          "type": "string",
          "minLength": 1
```

duncandewhurst commented on mapping/sustainability.yaml:
```diff
@@ -260,16 +260,16 @@
         Project level:
 
-        For each measure add a `ConservationMeasure` object to the `environment.conservationMeasures` array, setting its `.type` according to the matching code in the open "conservationMeasuresType" codelist and mapping the further explanation to its `.description`.
+        For each measure add a `ConservationMeasure` object to the `environment.conservationMeasures` array, set its `.type` according to the matching code in the open [conservationMeasure](../../reference/codelists.md#conservationmeasure) codelist and map the further explanation to its `.description`.
```
```suggestion
        For each measure add a `ConservationMeasure` object to the `environment.conservationMeasures` array, set its `.type` according to the open [conservationMeasure](../../reference/codelists.md#conservationmeasure) codelist and map the further explanation to its `.description`.
```

duncandewhurst reviewed (changes requested):
Happy for you to request a review from James once my comments are resolved.

odscjen committed:
Apply suggestions from code review

commit all suggestions except the one about `wholeListMerge` as waiting on outcome of another comment before deciding which route to go with that issue

Co-authored-by: Duncan Dewhurst <duncan.dewhurst@opendataservices.coop>

odscjen committed:
Merge branch '0.9-dev' into 401_conservation_measures

odscjen committed:
fix indent, update sustainability.md

odscjen committed:
Update schema/project-level/project-schema.json

Co-authored-by: Duncan Dewhurst <duncan.dewhurst@opendataservices.coop>

odscjen commented:
The error is just another example of no `id` field in an array object, but the test is to be changed as per https://github.com/open-contracting/infrastructure/pull/438#discussion_r1403805949

jpmckinney commented on schema/project-level/codelists/conservationMeasure.csv:
```diff
@@ -0,0 +1,11 @@
+setAsides,Set-asides,Identification of areas within the project boundaries that are excluded from development and targeted for conservation enhancement measures.
+invasiveSpeciesManagement,Invasive species management,"Measures taken to prevent the introduction of alien, invasive or non-native species, and combat the presence of such species identified within the project boundaries."
+habitatLossReduction,Habitat loss reduction,Measures that reduce the ecological habitats lost as a direct result of the project.
+pollutionReduction,Pollution reduction,"Measures that reduce the level of pollution generated as a direct result of the project. Pollution can include leakage into water courses or soil as well as noise, light, aerosol and gaseous emissions or vibrational nuisance to nearby wildlife populations."
```
```suggestion
pollutionReduction,Pollution reduction,"Measures that reduce the pollution generated as a direct result of the project. Pollution can include leakage into water courses or soil as well as noise, light, aerosol and gaseous emissions or vibrational nuisance to nearby wildlife populations."
```

jpmckinney commented on schema/project-level/codelists/conservationMeasure.csv:
```diff
@@ -0,0 +1,11 @@
+invasiveSpeciesManagement,Invasive species management,"Measures taken to prevent the introduction of alien, invasive or non-native species, and combat the presence of such species identified within the project boundaries."
+habitatLossReduction,Habitat loss reduction,Measures that reduce the ecological habitats lost as a direct result of the project.
+pollutionReduction,Pollution reduction,"Measures that reduce the level of pollution generated as a direct result of the project. Pollution can include leakage into water courses or soil as well as noise, light, aerosol and gaseous emissions or vibrational nuisance to nearby wildlife populations."
+soilManagement,Soil management,Measures that integrate sustainable soil management practices according to the Food and Agriculture Organization's Charter and Sustainable Soil Management Criteria.
```
```suggestion
soilManagement,Soil management,Measures that integrate sustainable soil management practices. For example, the Food and Agriculture Organization's Charter and Sustainable Soil Management Criteria.
```

jpmckinney commented on schema/project-level/codelists/documentType.csv:
```diff
@@ -64,6 +64,7 @@ contractAmendment,Contract amendment,A legal document that changes or adds to th
 designReport,Project design report,"A record of the justification for the design approach adopted in the project, typically issued before the works start, by the consultant or firm that will supervise the works.",OC4IDS
 paymentCertificate,Payment certificate,A document certifying that the work specified on a supplier's estimate or invoice is complete and that payment to the supplier is approved.,OC4IDS
 escalationApproval,Escalation approval,Documentation of one or more approved escalations of the contract price during the project life-cycle.,OC4IDS
+conservationMeasures,Conservation measures,Documentation of measures to be taken to mitigate the project's impact on biodiversity.,OC4IDS
```
The schema is "protect and enhance" rather than "mitigate the project's impact on"

odscjen committed:
Apply suggestions from code review

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

odscjen committed:
documentType.csv: apply review suggestion

odscjen committed:
Merge branch '0.9-dev' into 401_conservation_measures

odscjen committed:
fix merge errors in examples and conservationMeasure.csv

jpmckinney committed:
Merge branch '0.9-dev' into 401_conservation_measures

jpmckinney committed:
chore(consistency): Align document type code description with field description for conservationMeasures

jpmckinney reviewed (approved):
I committed a small change to align the documentType code description with the field description for conservationMeasures
